### PR TITLE
Set generic type arguments nullability for value types

### LINKED ISSUE #57920: Example usage of NullabilityInfoContext fails on preview.7
* 6.0.100-preview.7.21379.14
 * Current Window10
 * VS2019 Preview
  
here https://github.com/dotnet/core/issues/6444#issuecomment-891999111 the following (slightly modified) example is used 

    class Target
    {
        public (string?, object) TupleField;
    }

    [Fact]
    public void Test()
    {
        var type = typeof(Target);
        var tupleField = type.GetField("TupleField");

        var context = new NullabilityInfoContext();

        var tupleInfo = context.Create(tupleField);

        Assert.Equal(NullabilityState.NotNull, tupleInfo.ReadState);
        Assert.Equal(NullabilityState.Nullable, tupleInfo.GenericTypeArguments[0].ReadState);
        Assert.Equal(NullabilityState.NotNull, tupleInfo.GenericTypeArguments[1].ReadState);
    }

it fails since `tupleInfo.GenericTypeArguments` is empty.

I am not sure if this is a bug, or if the API has evolved beyond that example.

Paging @buyaa-n  since you posted that usage example

### PR DESCRIPTION
**Pull request by buyaa-n**
Fixes https://github.com/dotnet/runtime/issues/57920

Initially, I thought no need to check generic parameter for value types but turns out it needed

### CODE REVIEW

File: src/libraries/System.Runtime/tests/System/Reflection/NullabilityInfoContextTests.cs [hunk 1]
```diff
@@ -969,14 +1002,14 @@ internal class GenericTest<T>
         public Tuple<T?, string?, string?>? PropertyTupleNullNullNullNull { get; set; }
         public Tuple<T, T?, string> PropertyTupleNonNullNonNon { get; set; } = null!;
         Tuple<string?, T, T?>? PropertyTupleNullNonNullNull { get; set; }
-        public Tuple<T, string?, string>? PropertyTupleNonNullNonNull { get; set; }
-        public Tuple<string, string, T> PropertyTupleNonNonNonNon { get; set; } = null!;
+        public Tuple<T, int?, string>? PropertyTupleNonNullNonNull { get; set; }
+        public Tuple<int, string, T> PropertyTupleNonNonNonNon { get; set; } = null!;
```
Comment (buyaa-n) [inline src/libraries/System.Runtime/tests/System/Reflection/NullabilityInfoContextTests.cs:1006]: Found another bug when there is a value type included in the parameter list, the nullability of that value type is omitted from the nullability byte array, updated these test properties to test that fix

File: src/libraries/System.Private.CoreLib/src/System/Reflection/NullabilityInfoContext.cs [hunk 2]
```diff
@@ -368,32 +373,36 @@ private NullabilityInfo GetNullabilityInfo(MemberInfo memberInfo, Type type, ILi
...
 
-                    for (int i = 0; i < genericArguments.Length; i++)
+            if (underlyingType.IsGenericType)
+            {
+                Type[] genericArguments = underlyingType.GetGenericArguments();
+                genericArgumentsState = new NullabilityInfo[genericArguments.Length];
+
+                for (int i = 0, offset = 0; i < genericArguments.Length; i++)
+                {
+                    if (!genericArguments[i].IsValueType)
```
Comment (buyaa-n) [inline src/libraries/System.Private.CoreLib/src/System/Reflection/NullabilityInfoContext.cs:389]: when there is a value type included in the parameter list, the nullability of that value type is omitted from the nullability byte array

File: src/libraries/System.Runtime/tests/System/Reflection/NullabilityInfoContextTests.cs [hunk 3]
```diff
@@ -915,6 +944,11 @@ public class TypeWithNotNullContext
         private IDictionary<Type, string[]>? PropertyDictionaryNonNonNonNull { get; set; }
         public IDictionary<Type, string[]> PropertyDictionaryNonNonNonNon { get; set; } = null!;
 
+        public (string?, object) NullNonNonValueTuple;
```
[the remark this answers is not in the document]
Comment (steveharter) [reply src/libraries/System.Runtime/tests/System/Reflection/NullabilityInfoContextTests.cs:947]: Just a style comment, but perhaps Null_Non_Non_ValueTuple?

**Comment by steveharter** [comment]
Should this be ported to 6.0?

**Review by steveharter — APPROVED** [review]

File: src/libraries/System.Runtime/tests/System/Reflection/NullabilityInfoContextTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (buyaa-n) [reply src/libraries/System.Runtime/tests/System/Reflection/NullabilityInfoContextTests.cs]: updated

**Comment by buyaa-n** [comment]
> Should this be ported to 6.0?

Yes it is, was planning to do so after the PR review done

/backport to release/6.0

**Pull request merged**